### DISCUSSION TITLE
What does the global router and what is the relationship between global & detailed routers

### DISCUSSION THREAD

**Discussion by titan73** [The-OpenROAD-Project/OpenROAD#5602] [Q&A] [answered]
I understand what do the global & detailed placement as it can be seen visually but about the routing.
The detailed routing can be run without running global routing first so it's a bit confusing.

I'd say global routing prepares some information for detailed placement. What are this information?
I've heard about routing resource allocations. Also about "guides". What are this "guides"?

Is there some info that summarizes what the global route does?
And btw, how can the detailed router works without the global router being run first?

**Comment by maliberty** [comment, marked as answer]
The global router's job is to resolve routing congestion by giving general guidance to the detailed router.  That guidance is in the form of 'guides' which are layer-assigned rectangles that the net should be routed within.  If all nets are routed inside their guides the hope is that the detailed router will not hit an area that is too dense to route.

The detailed router then tries to implement the nets inside their guides considering the full set of DRC rules.  It is possible that it still can't find a solution within the guide and may deviate outside if necessary.

The detailed router in OR has a global router built into it.  We don't recommend you use it as it is rather untested and suggest you instead ues the global_route command.  It will be used if no guides are provided.

**Reply by titan73** [reply]
Ok. Thanks for the explanation.

**Comment by rovinski** [comment]
@titan73 here are some slides from my class on global vs. detailed routing. Hopefully it can add to what @maliberty said
[rovinski_ECE9433_routing.pdf](https://github.com/user-attachments/files/16686552/rovinski_ECE9433_routing.pdf)

**Reply by titan73** [reply]
Thanks for the doc.

**Comment by titan73** [comment]
Is there a way to visualize these guides?

**Reply by maliberty** [reply]
If you select a net in the inspector there is a guides button to draw the guides for that net.

**Reply by stefanottili** [reply]
After "read_lef", "read_def ispd19_test10/def/12.t10.def.gz", "global route", I can see "focus", "timing", "zoom to" and "highlight" button in the inspector widget.
Where is the "guides" button supposed to be ?
Is there another step necessary to visualize the global route guides ?

<img width="294" alt="image" src="https://github.com/user-attachments/assets/e86a6d8a-62ed-427c-ba9f-96a15219d23d">

**Reply by maliberty** [reply]
You should see something like:
![image](https://github.com/user-attachments/assets/00a6f81d-3790-4512-a551-004bd0a3e2d3)

with the 'Route Guides' button.  If there are no guides then that button is hidden.  Do you have a routed net selected?

**Reply by stefanottili** [reply]
I have a routed net selected, but it seems that reading a routed def followed by global_route won’t get me the guide button.

**Reply by maliberty** [reply]
If the net is already routed there is nothing for the global router to do and no guides are generated.

**Reply by titan73** [reply]
Thank.
For each inspected net, the guides for all these nets are displayed. Is there a way to clear the guides?

**Reply by maliberty** [reply]
You can click the guides button again to turn them off for the selected net.  If you right click in the layout there is a pop menu with the choice to clear all guides (and various other things).

**Reply by titan73** [reply]
Works fine. Thanks
